geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/AggregateRegionStatsMonitor.java

```diff
@@ -24,71 +26,118 @@
+    return monitors;
+  }
 
-  private Map<Statistics, MemberLevelRegionStatisticsListener> listeners;
+  Map<Statistics, MemberLevelRegionStatisticsListener> getListeners() {
+    return listeners;
+  }
 
   public AggregateRegionStatsMonitor(String name) {
     super(name);
-    monitors = new HashMap<Statistics, ValueMonitor>();
-    listeners = new HashMap<Statistics, MemberLevelRegionStatisticsListener>();
+    monitors = new HashMap<>();
+    listeners = new HashMap<>();
   }
 
-  @Override
-  public void addStatisticsToMonitor(Statistics stats) {
-    ValueMonitor regionMonitor = new ValueMonitor();
-    MemberLevelRegionStatisticsListener listener = new MemberLevelRegionStatisticsListener();
-    regionMonitor.addListener(listener);
-    regionMonitor.addStatistics(stats);
-    monitors.put(stats, regionMonitor);
-    listeners.put(stats, listener);
-  }
+  Number computeDelta(DefaultHashMap statsMap, String name, Number currentValue) {
+    if (name.equals(StatsKey.PRIMARY_BUCKET_COUNT)) {
+      Number prevValue = statsMap.get(StatsKey.PRIMARY_BUCKET_COUNT).intValue();
+      return currentValue.intValue() - prevValue.intValue();
+    }
 
+    if (name.equals(StatsKey.BUCKET_COUNT)) {
+      Number prevValue = statsMap.get(StatsKey.BUCKET_COUNT).intValue();
+      return currentValue.intValue() - prevValue.intValue();
+    }
 
-  public void removePartitionStatistics(Statistics stats) {
-    MemberLevelRegionStatisticsListener listener = removeListener(stats);
-    if (listener != null) {
-      listener.decreaseParStats(stats);
+    if (name.equals(StatsKey.TOTAL_BUCKET_SIZE)) {
+      Number prevValue = statsMap.get(StatsKey.TOTAL_BUCKET_SIZE).intValue();
+      return currentValue.intValue() - prevValue.intValue();
     }
-  }
 
-  public void removeLRUStatistics(Statistics stats) {
-    removeListener(stats);
-  }
+    if (name.equals(StatsKey.LRU_EVICTIONS)) {
+      Number prevValue = statsMap.get(StatsKey.LRU_EVICTIONS).longValue();
+      return currentValue.longValue() - prevValue.longValue();
+    }
 
-  public void removeDirectoryStatistics(Statistics stats) {
-    removeListener(stats);
+    if (name.equals(StatsKey.LRU_DESTROYS)) {
+      Number prevValue = statsMap.get(StatsKey.LRU_DESTROYS).longValue();
+      return currentValue.longValue() - prevValue.longValue();
+    }
+
+    if (name.equals(StatsKey.DISK_SPACE)) {
+      Number prevValue = statsMap.get(StatsKey.DISK_SPACE).longValue();
+      return currentValue.longValue() - prevValue.longValue();
+    }
+
+    return 0;
   }
 
-  @Override
-  public void stopListener() {
-    for (Statistics stat : listeners.keySet()) {
-      ValueMonitor monitor = monitors.get(stat);
-      monitor.removeListener(listeners.get(stat));
-      monitor.removeStatistics(stat);
+  void increaseStats(String name, Number value) {
+    if (name.equals(StatsKey.PRIMARY_BUCKET_COUNT)) {
+      primaryBucketCount.set(primaryBucketCount.get() + value.intValue());
```

Comment:
This has the same race that the old code had -- these should be `getAndAdd` calls.
Also, is there a reason we're using `intValue` instead of `longValue`?